app/tests/integration/nimbus/pages/experimenter/summary.py

```diff
@@ -155,3 +179,17 @@ def clone_save(self):
     def clone(self):
         self.clone_action.click()
         self.clone_save.click()
+
+    @property
+    def promote_to_rollout_buttons(self):
+        self.wait.until(
+            EC.presence_of_all_elements_located(self._clone_action_locator),
+            message="Summary Page: could not find promote to rollout buttons",
+        )
+        return self.find_elements(*self._promote_rollout_locator)
+
+    def promote_first_branch_to_rollout(self):
+        self.promote_to_rollout_buttons[0].click()
+        random_id = str(random())
+        self.clone_name = f"{self.experiment_slug}-rollout-clone-{random_id}"
+        self.clone_save.click()
```

Comment:
Opening as a draft while I keep banging on this to see if I can make this integration test not use a dirty random number. Tried grabbing the experiment slug from the DOM to use as a unique ID prefix, but the property just comes up as an empty string.